src/nvim/terminal.c

```diff
@@ -793,6 +794,57 @@ static VTermKey convert_key(int key, VTermModifier *statep)
     case K_KMULTIPLY: return VTERM_KEY_KP_MULT;
     case K_KDIVIDE:   return VTERM_KEY_KP_DIVIDE;
 
+    case K_S_F1:      *statep |= VTERM_MOD_SHIFT;
```

Comment:
I thought about that, but if we took that approach, we'd still need to check all the `K_S_*` keys individually in the big switch since the `K_S_*` keys count as different keys from the `K_*` ones.

If I understand your suggestion correctly, such a function would only let us go from
```
case K_S_F1:      *statep |= VTERM_MOD_SHIFT;
case K_F1:        return VTERM_KEY_FUNCTION(1);
```
to
```
case K_S_F1:
case K_F1:        return VTERM_KEY_FUNCTION(1);
```